RefactoringPracticeI/GameInventory/Inventory/GameInventory.cs

```diff
@@ -11,66 +11,21 @@ public GameInventory(IList<Item> Items)
 
         public void UpdateQuality()
         {
-            for (var index = 0; index < _items.Count; index++)
+            foreach (Item item in _items)
             {
-                if (_items[index].Name != "Aged Brie" && _items[index].Name != "Backstage passes to a Pokemon Gym concert")
-                {
-                    if (_items[index].Quality > 0 && _items[index].Name != "Sulfuras, Hand of Ragnaros")
-                    {
-                        _items[index].Quality--;
-                    }
-                }
-                else
-                {
-                    if (_items[index].Quality < 50)
-                    {
-                        _items[index].Quality++;
-
-                        if (_items[index].Name == "Backstage passes to a Pokemon Gym concert")
-                        {
-                            if (_items[index].SellIn < 11 && _items[index].Quality < 50)
-                            {
-                                _items[index].Quality++;
-                            }
-
-                            if (_items[index].SellIn < 6 && _items[index].Quality < 50)
-                            {
-                                _items[index].Quality++;
-                            }
-                        }
-                    }
-                }
+                Item result;
 
-                if (_items[index].Name != "Sulfuras, Hand of Ragnaros")
+                result = item.Name switch
```

Comment:
Nice, most people states that using a "switch" statement is a bad practice, have not been able to validate that, to me still its a nice tool, but you might encounter people suggesting to change this to if/else, that is subjective, unless strong arguments back it up.